modules/data-entry/src/main/frontend/src/dataHomepage/FilterComponents/QuestionnaireFilter.jsx

```diff
@@ -0,0 +1,131 @@
+//
+//  Licensed to the Apache Software Foundation (ASF) under one
+//  or more contributor license agreements.  See the NOTICE file
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the
+//  "License"); you may not use this file except in compliance
+//  with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+
+import React, { forwardRef, useState } from "react";
+import { Select, MenuItem, Card, CardHeader, CardContent, withStyles } from "@material-ui/core";
+import ErrorIcon from "@material-ui/icons/Error";
+import PropTypes from "prop-types";
+
+import FilterComponentManager from "./FilterComponentManager.jsx";
+import { DEFAULT_COMPARATORS, UNARY_COMPARATORS } from "./FilterComparators.jsx";
+import QuestionnaireStyle from "../../questionnaire/QuestionnaireStyle.jsx";
+
+const COMPARATORS = DEFAULT_COMPARATORS.slice().concat(UNARY_COMPARATORS);
+
+/**
+ * Display a filter on the associated questionnaire of a form. This is not meant to be instantiated directly, but is returned from FilterComponentManager's
+ * getFilterComparatorsAndComponent method.
+ *
+ * @param {string} defaultValue The default value to place in the questionnaire filter
+ * @param {func} onChangeInput Function to call when this filter has chosen a new questionnaire
+ * @param {func} questionDefinition Unused, here to stop a warning when it is passed to the SearchBar component
+ * Other props will be forwarded to the SearchBar component
+ */
+const QuestionnaireFilter = forwardRef((props, ref) => {
+  const { classes, defaultValue, onChangeInput, questionDefinition, ...rest } = props;
+  const [ error, setError ] = useState();
+  // Store information about each questionnaire and whether or not we have
+  // initialized
+  let [ questionnaires, setQuestionnaires ] = useState([]);
+  let [ initialized, setInitialized ] = useState(false);
+  // Store selected questionnaire uuid
+  let [ selection, setSelection ] = useState(defaultValue);
+  let [ uuidToTitle, setUuidToTitle ] = useState({});
+
+  let mapObject = {};
+  // Obtain information about the questionnaires available to the user
+  let initialize = () => {
+    setInitialized(true);
+
+    // Fetch the questionnaires
+    fetch("/query?query=select * from [lfs:Questionnaire]")
```

Comment:
This only fetches 10 questionnaires.

```suggestion
    fetch("/query?limit=100&query=select * from [lfs:Questionnaire]")
```

If there are more than 100 questionnaires, there are bigger issues to solve, and we'll have to replace this with a typeahead input.